src/MapViewController.swift

```diff
@@ -417,11 +476,184 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
+      return
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:didRecognizeSingleTapGesture:))))! else {
+      return
+    }
+    self.gestureDelegate?.mapController!(self, recognizer: recognizer, didRecognizeSingleTapGesture: location)
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, shouldRecognizeDoubleTapGesture location: CGPoint) -> Bool {
+    guard (self.gestureDelegate != nil) else {
+      return true
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:shouldRecognizeDoubleTapGesture:))))! else {
+      return true
+    }
+    return ((self.gestureDelegate?.mapController!(self, recognizer: recognizer, shouldRecognizeDoubleTapGesture: location))!)
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, didRecognizeDoubleTapGesture location: CGPoint) {
+    guard (self.gestureDelegate != nil) else {
+      return
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:didRecognizeDoubleTapGesture:))))! else {
+      return
+    }
+    self.gestureDelegate?.mapController!(self, recognizer: recognizer, didRecognizeDoubleTapGesture: location)
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, shouldRecognizeLongPressGesture location: CGPoint) -> Bool {
+    guard (self.gestureDelegate != nil) else {
+      return true
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:shouldRecognizeLongPressGesture:))))! else {
+      return true
+    }
+    return (self.gestureDelegate?.mapController!(self, recognizer: recognizer, shouldRecognizeLongPressGesture: location))!
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, didRecognizeLongPressGesture location: CGPoint) {
+    guard (self.gestureDelegate != nil) else {
+      return
+    }
+    guard (self.gestureDelegate?.responds(to: #selector(MapGestureDelegate.mapController(_:recognizer:didRecognizeLongPressGesture:))))! else {
+      return
+    }
+    self.gestureDelegate?.mapController!(self, recognizer: recognizer, didRecognizeLongPressGesture: location)
+  }
+  
+  open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, shouldRecognizePanGesture displacement: CGPoint) -> Bool {
+    if (!panEnabled) {
+      self.shouldFollowCurrentLocation = false
+      self.findMeButton.isSelected = false
+    }
+    return panEnabled
+  }
+  
   open func mapView(_ view: TGMapViewController, recognizer: UIGestureRecognizer, didRecognizePanGesture location: CGPoint) {
-    shouldFollowCurrentLocation = false
-    findMeButton.isSelected = false
+    self.shouldFollowCurrentLocation = false
+    self.findMeButton.isSelected = false
+    guard (self.panDelegate != nil) else {
```

Comment:
If this is all you're doing here, you could rewrite this like:
```
if let panDelegate = panDelegate {
  panDelegate.mapController(self, didPanMap: location)
}
```
Swift will figure out to use the shadow unwrapped optional. I generally don't like doing this for guard statements but the if/let syntax makes it very obvious which you're using as it scopes the shadowed variable to just inside it's curlies.